src/routes/addMovies.ts

```diff
@@ -0,0 +1,57 @@
+import config from '../config';
+import { FastifyRequest } from 'fastify';
+import { StatusCodes, getReasonPhrase } from 'http-status-codes';
+import { IStorage } from '../storage';
+
+
+export default {
+    init: (httpClient: any, storage: IStorage) => {
+        if ( !httpClient || !storage ) {
+            throw new Error('Router dependencies are incorrect.');
+        }
+
+        return {
+            schema: {
+                body: {
+                    // name of the movie, overwrites original film title if given
+                    name: {
+                        type: 'string'
+                    },
+                    // user's comment
+                    comment: {
+                        type: 'string'
+                    },
+                    // user's movie score
+                    personalScore: {
+                        type: 'number'
+                    }
+                }
+            },
+            handler: async function ( request: FastifyRequest ) {
+                try {
+                    // @ts-ignore
+                    const { name, comment, personalScore } = request.body;
+
+                    let { data: info } = await httpClient.get(`${config.OMDB_API_BASE_URL}&t=${name}`);
```

Comment:
usually interaction with third parties are not done within controller handler. Better to create a separate service for communicating with omdbapi

Reply:
also in this service in one place we do all error handling and mapping related to this service

Reply:
Done.